validator/accounts/account.go

```diff
@@ -260,3 +260,20 @@ func homeDir() string {
 	}
 	return ""
 }
+
+// ExtractPublicKeysFromKeyStore extracts only the public keys from the decrypted keys from the keystore.
+func ExtractPublicKeysFromKeyStore(keystorePath string, passphrase string) ([][]byte, error) {
+	decryptedKeys, err := DecryptKeysFromKeystore(keystorePath, params.BeaconConfig().ValidatorPrivkeyFileName, passphrase)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Could not decrypt keys from keystore in path %s", keystorePath)
+	}
+
+	i := 0
```

Comment:
what is i for ? , this is not needed, you can look at the index of the array in the loop itself

Reply:
`decryptedKeys` is a map of `map[string]*keystore.Key`. Frist value is a string, not an index.

Reply:
Makes sense, thanks for clarifying